lib/webpack/entry-points-plugin.js

```diff
@@ -0,0 +1,132 @@
+/*
+ * This file is part of the Symfony Webpack Encore package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
+const copyEntryTmpName = require('../utils/copyEntryTmpName');
+
+/**
+ * Return the file extension from a filename, without the leading dot and without the query string (if any).
+ *
+ * @param {string} filename
+ * @returns {string}
+ */
+function getFileExtension(filename) {
+    return path.extname(filename).slice(1).split('?')[0];
+}
+
+class EntryPointsPlugin {
+    /**
+     * @param {String} publicPath The public path of the assets, from where they are served
+     * @param {String} outputPath The output path of the assets, from where they are saved
+     * @param {Array<String>} integrityAlgorithms The algorithms to use for the integrity hash
+     */
+    constructor({
+        publicPath,
+        outputPath,
+        integrityAlgorithms
+    }) {
+        this.publicPath = publicPath;
+        this.outputPath = outputPath;
+        this.integrityAlgorithms = integrityAlgorithms;
+    }
+
+    apply(compiler) {
+        compiler.hooks.afterEmit.tapAsync({ name: 'EntryPointsPlugin' }, (compilation, callback) => {
+            const manifest = {
+                entrypoints: {},
+            };
+
+            const stats = compilation.getStats().toJson({
+                assets: true,
+                moduleAssets: true,
+                relatedAssets: false,
+                chunkGroupAuxiliary: false,
+                chunks: false,
+                modules: false,
+                timings: false,
+                logging: false,
+                errorDetails: false,
+            });
+
+            for (const [entryName, entry] of Object.entries(stats.entrypoints)) {
+                // We don't want to include the temporary entry in the manifest
+                if (entryName === copyEntryTmpName) {
+                    continue;
+                }
+
+                manifest.entrypoints[entryName] = {};
+
+                for (const asset of entry.assets) {
+                    // We don't want to include hot-update files in the manifest
+                    if (asset.name.includes('.hot-update.')) {
+                        continue;
+                    }
```

Comment:
When using the dev-server and you update a file, an `.hot-update.` file can be emitted, but we don't want to add it in the manifest.